### DISCUSSION TITLE
Foundation model finetuning

### DISCUSSION THREAD

**Discussion by douniashaabankabakibo** [ACEsuit/mace#821] [Q&A]
Hello,
I'm currently trying to fine-tune mace_agnesi_medium.model, but I'm encountering convergence issues.
I began by running a VASP molecular dynamics (MD) simulation on my target material for 25,000 steps with a 2 fs timestep at T=1000K. From this trajectory, I randomly selected 100 frames for training. However, the RMSE_E and RMSE_F are not decreasing sufficiently.
I've experimented with various hyperparameters, but none have significantly improved the results. Could the issue be with my fine-tuning approach, or is it related to the training set? The structure was not relaxed before the MD run—I used an experimentally measured structure. Intuitively, I wouldn’t expect this to impact fine-tuning as much, but I might be wrong?
I’ve attached the input files and log for reference.

Thanks in advance for your help!
Dounia

[finetuning.zip](https://github.com/user-attachments/files/18737151/finetuning.zip)

**Comment by birdyLinch** [comment]
Hi there, through reproducing your experiment, I believe you can fix it by puting the --forces_key=forces to your finetune script.

> 2025-02-11 22:52:49.579 INFO: ===========TRAINING===========
> 2025-02-11 22:52:49.579 INFO: Started training, reporting errors on validation set
> 2025-02-11 22:52:49.579 INFO: Loss metrics on validation set
> 2025-02-11 22:52:59.434 INFO: Initial: head: default, loss=0.68870832, RMSE_E_per_atom= 2614.84 meV, RMSE_F=   70.51 meV / A
> ^B^[[5~^[[5~2025-02-11 22:53:16.062 INFO: Epoch 0: head: default, loss=0.00689844, RMSE_E_per_atom=    6.57 meV, RMSE_F=   83.03 meV / A
> 2025-02-11 22:53:18.752 INFO: Epoch 1: head: default, loss=0.00351799, RMSE_E_per_atom=    1.36 meV, RMSE_F=   59.31 meV / A
> 2025-02-11 22:53:21.447 INFO: Epoch 2: head: default, loss=0.00223972, RMSE_E_per_atom=    2.61 meV, RMSE_F=   47.32 meV / A
> 2025-02-11 22:53:24.135 INFO: Epoch 3: head: default, loss=0.00166926, RMSE_E_per_atom=    0.90 meV, RMSE_F=   40.86 meV / A
> 2025-02-11 22:53:26.828 INFO: Epoch 4: head: default, loss=0.00133940, RMSE_E_per_atom=    1.94 meV, RMSE_F=   36.59 meV / A
> 2025-02-11 22:53:29.536 INFO: Epoch 5: head: default, loss=0.00112104, RMSE_E_per_atom=    0.66 meV, RMSE_F=   33.48 meV / A
> 2025-02-11 22:53:32.245 INFO: Epoch 6: head: default, loss=0.00096544, RMSE_E_per_atom=    0.81 meV, RMSE_F=   31.07 meV / A
> 2025-02-11 22:53:34.932 INFO: Epoch 7: head: default, loss=0.00085105, RMSE_E_per_atom=    0.92 meV, RMSE_F=   29.17 meV / A
> 2025-02-11 22:53:37.625 INFO: Epoch 8: head: default, loss=0.00076567, RMSE_E_per_atom=    0.83 meV, RMSE_F=   27.67 meV / A
> 2025-02-11 22:53:40.301 INFO: Epoch 9: head: default, loss=0.00070553, RMSE_E_per_atom=    0.72 meV, RMSE_F=   26.56 meV / A
> 2025-02-11 22:53:43.047 INFO: Epoch 10: head: default, loss=0.00065516, RMSE_E_per_atom=    0.49 meV, RMSE_F=   25.60 meV / A

**Comment by douniashaabankabakibo** [comment]
Hello,
Thank you for the quick response. That did the trick—I hadn’t realized that --forces_key=forces wasn’t set by default. Much appreciated!

**Comment by gabor1** [comment]
Also look at the warning issued by the mace training script: using "forces" as the name of the key is _unsafe_ because of changes in ASE. we recommend that you come up with your own key, and specify it explicitly. We are not forcing people to do this in order to have some backwards compatibility.

**Comment by douniashaabankabakibo** [comment]
Will do. Thank you for pointing that out!
